### PR TITLE
Bug 1209978 - [Settings] CSS refactoring

### PR DESCRIPTION
**Pull request by fabi1cazenave**
https://bugzilla.mozilla.org/show_bug.cgi?id=1209978

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=60319f6ba4267e1fc5b85135017b8e4f3be9a073

File: apps/settings/style/downloads.css [hunk 1]
```diff
@@ -93,20 +93,17 @@
 
 /* Text */
 .fileName {
-  padding-top: 1rem;
-  -moz-padding-end: 6rem;
-  padding-bottom: 1rem;
-  -moz-padding-start: 3.5rem;
-  -moz-box-sizing: border-box;
+  padding: 1rem 6rem;
+  box-sizing: border-box;
```
Comment (gasolin) [inline apps/settings/style/downloads.css]: missing `padding-inline-start: 3.5rem;`?

File: apps/settings/style/icons.css [hunk 2]
```diff
@@ -260,7 +261,7 @@ aside.connecting {
  */
 
 .privacy-menuitem {
```
Comment (gasolin) [inline apps/settings/style/icons.css]: Privacy control panel is removed, so we can remove this style as well

File: apps/settings/style/downloads.css (hunk 1 above)
Comment (fabi1cazenave) [inline apps/settings/style/downloads.css]: ugh, good catch

File: apps/settings/style/icons.css (hunk 2 above)
Comment (fabi1cazenave) [inline apps/settings/style/icons.css]: ok

File: apps/settings/style/downloads.css [hunk 3]
```diff
@@ -212,40 +209,34 @@
...
+html[dir="rtl"] #downloadList ul > li[data-state="failed"] .pack-end {
+  background-position: right center;
 }
 
-html:-moz-dir(rtl) #downloads-edit-button::before {
-  transform: initial;
+#downloadList ul > li[data-state="succeeded"] .pack-end {
+  display: none;
 }
+
```
Comment (gasolin) [inline apps/settings/style/downloads.css]: remove extra line

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f8311cc51202ee09b3e7d8e2486a780b96e5a818

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6a3a48998111dc7d2aec080140d4ad9d6731b2cd

File: apps/settings/style/downloads.css (hunk 1 above)
Comment (fabi1cazenave) [inline apps/settings/style/downloads.css]: oh. No. This has been replaced by the two lines below:
```css
html[dir="ltr"] .fileName { padding-left: 3.5rem; }
html[dir="rtl"] .fileName { padding-right: 3.5rem; }
```

These two lines work even if the `.fileName` element has a `dir="auto"` attribute — and it needs one to display LTR file names in an RTL context.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d214740784352961bb04329bd629c2c784c39765

File: apps/settings/style/settings.css [hunk 4]
```diff
@@ -918,13 +923,12 @@ ul li > label.range-icons.volume input {
   height: auto;
 }
 
-#feedback-happy {
-  float: left;
-}
+html[dir="ltr"] #feedback-happy { float: left; }
```
Comment (gasolin) [inline apps/settings/style/settings.css]: might worth to add a comment /* can replace with float: inline-end once its support */

File: apps/settings/style/downloads.css (hunk 3 above)
Comment (fabi1cazenave) [inline apps/settings/style/downloads.css]: done

File: apps/settings/style/settings_large.css [hunk 5]
```diff
@@ -65,10 +65,15 @@
...
-    background: url("images/ui/tablet_separator.png") no-repeat right / 0.2rem transparent;
+    background-image: url("images/ui/tablet_separator.png");
+    background-repeat: no-repeat;
+    background-size: 0.2rem;
+    background-color: transparent;
     height: 95%;
     top: 0;
   }
+  html[dir="ltr"] section[role="region"]#root header:after { background-position: right; }
+  html[dir="rtl"] section[role="region"]#root header:after { background-position: left; }
```
Comment (gasolin) [inline apps/settings/style/settings_large.css]: does `background-position` will support `start/end` or to be tracked in 
https://wiki.mozilla.org/Gaia/CSS_Guidelines#Direction_.28RTL.2FLTR_and_BiDi.29 ?

File: apps/settings/style/settings.css (hunk 4 above)
Comment (fabi1cazenave) [inline apps/settings/style/settings.css]: I’ll add a reference to https://bugzilla.mozilla.org/show_bug.cgi?id=1122918 while I’m at it

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=83a8bea26f258426b38e6a552c931f7490b49a90

File: apps/settings/style/downloads.css (hunk 1 above)
Comment (gasolin) [inline apps/settings/style/downloads.css]: thanks for clarify. maybe it's worth to note at https://wiki.mozilla.org/Gaia/CSS_Guidelines#Direction_.28RTL.2FLTR_and_BiDi.29

File: apps/settings/style/settings_large.css (hunk 5 above)
Comment (fabi1cazenave) [inline apps/settings/style/settings_large.css]: I’m not aware of any work or spec to support `start/end` in `background-position`. If you have a reference, I’ll be happy to track it.

File: apps/settings/style/downloads.css (hunk 1 above)
Comment (fabi1cazenave) [inline apps/settings/style/downloads.css]: that’s already mentioned at the end of this wiki page

File: apps/settings/style/icons.css [hunk 6]
```diff
@@ -253,82 +255,3 @@ aside.connecting {
...
-  width: 3.2rem;
-  height: 3.3rem;
-  background: url(/shared/resources/branding/privacy_sprite.png) no-repeat 3rem 3rem / 9.6rem;
-  position: absolute;
-  left: 1.5rem;
-  top: 50%;
-  margin-top: -1.5rem;
-}
-
-.privacy-browserOS:before {
```
Comment (gasolin) [inline apps/settings/style/icons.css]: It's not related to privacy panel... it's been used in `about_your_privacy.html` so we need keep it

File: apps/settings/style/icons.css [hunk 7]
```diff
@@ -253,82 +255,3 @@ aside.connecting {
...
-  background-position: 0 0;
-}
-
-.privacy-marketPlace:before {
```
Comment (gasolin) [inline apps/settings/style/icons.css]: we need keep it as well

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d3519e446aa6573045674089c4e52e251793a950

File: apps/settings/style/icons.css (hunk 6 above)
Comment (fabi1cazenave) [inline apps/settings/style/icons.css]: oops. Sorry for the misunderstanding. Fixed.

File: apps/settings/style/icons.css (hunk 7 above)
Comment (fabi1cazenave) [inline apps/settings/style/icons.css]: fixed

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0763d35d487f9f18714353187be744cfd914c639

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0d5c4c11f7a1b5339110c6410722032fcf635127

**Comment by fabi1cazenave** [comment]
Just squashed all commits now that the review is done. Thanks!

**Pull request merged**